### PR TITLE
Fixes #8432: Missing an information line in rule details to explain that the graphe is clickable

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8432

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
On first acces to that page, last period should be selected so you shopuld update the default text and highlight the last period (the one on the right)

**Comment by RaphaelGauthier** [comment]
PR updated

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 1]
```diff
@@ -190,6 +210,14 @@ class RuleCompliance (
                 x: { show: true }
               , y: { show: true }
             }
+          , onrendered: function () {
+              if(firstLoad){
+                  document.getElementsByClassName('c3-bar-12')[0].classList.add('c3-bar-highlighted');
+                  $$('#selectedPeriod b').text(x[x.length-1]);
+                  console.log('ok');
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: console.log here :)

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 2]
```diff
@@ -190,6 +210,14 @@ class RuleCompliance (
...
+                  firstLoad = false;
+              }
+          }
         } );
         $$('#changesChart').html(chart.element);
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: Maybe you should do:

```
document.getElementsByClassName('c3-bar-12')[0].classList.add('c3-bar-highlighted');
$$('#selectedPeriod b').text(x[x.length-1]);
```

and remove the onrendered function and the firstload var

Comment (RaphaelGauthier) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: I tried but it didn't work. In this case :
`document.getElementsByClassName('c3-bar-12')[0]` 
returns an undefined object... even if I call it after the chart generation...

**Comment by RaphaelGauthier** [comment]
PR updated

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala (hunk 2 above)
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: Have you tried the 'oninit' callback ? Which is called only once when chart is initialized. Maybe you could get rid of the firstload variable that i don't like :)

http://c3js.org/reference.html#oninit

Comment (RaphaelGauthier) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: I don't really like it too... But it doesn't work with the 'oninit' callback too... :/

**Comment by RaphaelGauthier** [comment]
PR updated

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 3]
```diff
@@ -190,6 +211,11 @@ class RuleCompliance (
                 x: { show: true }
               , y: { show: true }
             }
+          , onrendered: function () {
+                  document.getElementsByClassName('c3-bar-'+(selectedIndex).toString())[0].classList.add('c3-bar-highlighted');
+                  $$('#selectedPeriod b').text(x[selectedIndex+1]);
+                  firstLoad = false;
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: remove firstLoad here

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 4]
```diff
@@ -167,14 +179,23 @@ class RuleCompliance (
...
         x.splice(0,0,'x')
+        var selectedIndex = x.length-2;
         var chart = c3.generate({
           data: {
                 x: 'x'
               , columns: [ x , data ]
               , type: 'bar'
               , onclick: function (d, element) {
+                  selectedIndex = d.index;
                   ${SHtml.ajaxCall(JsRaw("recentChanges.t[d.index]"),  s => refreshTableChanges(Some(s.toLong)))}
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: change d.index to 'selectedIndex'

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 5]
```diff
@@ -167,14 +179,23 @@ class RuleCompliance (
...
+                  refreshInterval(x[d.index+1],element);
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: d.index to -> selectedIndex

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala [hunk 6]
```diff
@@ -190,6 +211,11 @@ class RuleCompliance (
                 x: { show: true }
               , y: { show: true }
             }
+          , onrendered: function () {
+                  document.getElementsByClassName('c3-bar-'+(selectedIndex).toString())[0].classList.add('c3-bar-highlighted');
+                  $$('#selectedPeriod b').text(x[selectedIndex+1]);
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala]: Is it possible to replace those two lines by:

```
var element = document.getElementsByClassName('c3-bar-'+(selectedIndex).toString())[0]
refreshInterval(x[selectedIndex+1],element);
```

So you use the same code

File: rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js [hunk 7]
```diff
@@ -1314,6 +1314,18 @@ function refreshTable (gridId, data) {
   table.draw();
 }
 
+function refreshInterval(interval, element){
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/rudder-datatable.js]: You could have named this function 'selectInterval'

**Comment by RaphaelGauthier** [comment]
PR updated

**Comment by RaphaelGauthier** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**